lib/licensed/sources/nuget.rb

```diff
@@ -0,0 +1,230 @@
+
+        def initialize(name:, version:, path:, search_root: nil, metadata: {}, errors: [])
+          super(name: name, version: version, path: path, search_root: search_root, metadata: metadata, errors: errors)
+          @metadata["homepage"] = project_url if project_url
+        end
+
+        def nuspec_path
+          name = @metadata["name"]
+          File.join(self.path, "#{name}.nuspec")
+        end
+
+        def nuspec_contents
+          return unless nuspec_path
+          @nuspec_contents ||= File.read(nuspec_path)
+        end
+
+        def project_url
+          return unless nuspec_contents
+          @project_url ||= begin
+            match = nuspec_contents.match PROJECT_URL_REGEX
+            match[1] if match && match[1]
+          end
+        end
+
+        def project_files
+          @nuget_project_files ||= begin
+            files = [super(), nuspec_local_license_file].flatten.compact
+
+            # Only download licenseUrl if no recognized license was found locally
+            if files.none? { |file| file.license && file.license.key != "other" }
+              files.push(nuspec_remote_license_file)
+            end
+            files.compact
+          end
+        end
+
+        # Look for a <license type="file"> element in the nuspec that points to an
+        # on-disk license file (which licensee may not find due to a non-standard filename)
+        def nuspec_local_license_file
+          return @nuspec_local_license_file if defined?(@nuspec_local_license_file)
+          return unless nuspec_contents
+
+          match = nuspec_contents.match LICENSE_FILE_REGEX
+          return unless match && match[1]
+
+          license_path = File.join(File.dirname(nuspec_path), match[1])
+          return unless File.exist?(license_path)
+
+          license_data = File.read(license_path)
+          @nuspec_local_license_file = Licensee::ProjectFiles::LicenseFile.new(license_data, license_path)
+        end
+
+        # Look for a <licenseUrl> element in the nuspec that either is known to contain a license identifier
+        # in the URL, or points to license text on the internet that can be downloaded.
+        def nuspec_remote_license_file
+          return @nuspec_remote_license_file if defined?(@nuspec_remote_license_file)
+          return unless nuspec_contents
+
+          match = nuspec_contents.match LICENSE_URL_REGEX
+          return unless match && match[1]
+
+          # Attempt to fetch the license content
+          license_content = self.class.retrieve_license(match[1])
+          @nuspec_remote_license_file = Licensee::ProjectFiles::LicenseFile.new(license_content, { uri: match[1] }) if license_content
+        end
+
+        class << self
+          def strip_html(html)
+            return unless html
+
+            if html.downcase.include?("<html")
+              ReverseMarkdown.convert(html, unknown_tags: :bypass)
+            else
+              html
+            end
```

Comment:
ruby idiom is to use a guard clause 👍 
```suggestion
            return html unless html.downcase.include?("<html")
            ReverseMarkdown.convert(html, unknown_tags: :bypass)
```